core/trino-main/src/main/java/io/trino/sql/planner/optimizations/AddExchanges.java

```diff
@@ -276,7 +277,7 @@ else if ((!isStreamPartitionedOn(child.getProperties(), partitioningRequirement)
          */
         private Optional<List<Symbol>> useParentPreferredPartitioning(AggregationNode node, Set<Symbol> parentPreferredPartitioningColumns)
         {
-            if (isUseExactPartitioning(session)) {
+            if (!isUseCostBasedPartitioning(session)) {
```

Comment:
We should still check for `isUseExactPartitioning(session)` right ? If use_exact_partitioning is true, I think that should supercede the new flag.

Reply:
ok, done